presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -512,7 +512,7 @@ private ConnectorTableMetadata getTableMetadata(ConnectorSession session, Schema
     private ConnectorTableMetadata doGetTableMetadata(ConnectorSession session, SchemaTableName tableName)
     {
         Optional<Table> table = metastore.getTable(new HiveIdentity(session), tableName.getSchemaName(), tableName.getTableName());
-        if (!table.isPresent() || table.get().getTableType().equals(TableType.VIRTUAL_VIEW.name())) {
+        if (!table.isPresent() || (!translateHiveViews && isHiveOrPrestoView(table.get()))) {
```

Comment:
Add tests (it is a pity that we only have product tests for translated Hive views).